### ISSUE TITLE
oauth app creation resource for github

### ISSUE THREAD

**Issue by sharmavijay86** [integrations/terraform-provider-github#786]
Hey there,
We are working on some frontend react integrations with terraform code. Where in we want to provision application and then create on github oauth application. All above is with terraform. So that now we can incorporate this oauth details with application to make everything automated, from oauth app creation to web app creation to its integration etc.

I don't see the resource type in documentation, so any way around.??

**Comment by jcudit** [comment]
Seems like we'd need a new resource to satisfy your use case.  We currently do not have a way to create apps via the provider.

**Comment by Kampe** [comment]
Would love this as well, this is one of the last things we need to automate within github.

**Comment by mike-kaminski** [comment]
This would be great to implement inline w/ the https://docs.github.com/en/rest/reference/apps#create-a-github-app-from-a-manifest API functionality.

**Comment by kory33** [comment]
We also have an use-case where an automatic creation of OAuth Apps comes handy as we are trying to declare everything in our infrastructure.

I am happy to work on this issue. However, what @mike-kaminski has suggested above

> This would be great to implement inline w/ the https://docs.github.com/en/rest/reference/apps#create-a-github-app-from-a-manifest API functionality.

only works for a GitHub App creation. While I am also interested in creating GitHub Apps automatically (I might open another issue for this), IIUC the manifest API cannot solve the original problem in this issue.

Moreover, I could not find a feature both in go-github library nor in the official GitHub REST API to create an OAuth App directly. At the current time this looks like an upstream issue, but any other thoughts on this?

**Comment by riuvshyn** [comment]
also looking forward to this

**Comment by ClaytonOlleyNutrien** [comment]
I'd love to see this implemented as well!

**Comment by gthieleb** [comment]
Any news on that? I would like to have this implemented to create rancher clusters with terraform where user can authenticate via github. Is the go-api a showstopper here?

**Comment by kfcampbell** [comment]
It's currently waiting on upstream support. We would need this feature available in the GitHub API before we could work on it at the Terraform provider level.

**Comment by markti** [comment]
> > It's currently waiting on upstream support. We would need this feature available in the GitHub API before we could work on it at the Terraform provider level.
> 
> Is this supported by [google/go-github#1335](https://github.com/google/go-github/pull/1335) or is there more functionality needed?

agree it does seem like this would be enough.

**Comment by markti** [comment]
Status: Pinned

**Comment by riuvshyn** [comment]
can't believe this is still not implemented... it's been 4years of waiting to stop doing things manually... very frustrating

**Comment by deiga** [comment]
Hey @riuvshyn ,
Please notice that there is no API for creating OAuth Apps and hence the Terraform provider can't implement a resource to do that.
